src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -58,31 +68,122 @@ function useFetchCounterparties () {
   }
 }
 
+const VIEW = {
+  pending: 'pending',
+  recent: 'recent'
+}
+
+const presentTruncatedAmount = (string, number = 15) => {
+  if (string.length > number) return `${sliceHash(string, number)}...`
+  return sliceHash(string, number)
+}
+
 export default function Inbox () {
-  const { data: { holofuelActionableTransactions: transactions = [] } = {} } = useQuery(HolofuelActionableTransactionsQuery)
+  const { data: { holofuelUser: whoami = {} } = {} } = useQuery(HolofuelUserQuery)
+  const { data: { holofuelLedger: { balance: holofuelBalance } = { balance: 0 } } = {} } = useQuery(HolofuelLedgerQuery)
+  const { data: { holofuelActionableTransactions: actionableTransactions = [] } = {} } = useQuery(HolofuelActionableTransactionsQuery)
+  const { data: { holofuelRecentTransactions: recentTransactions = [] } = {} } = useQuery(HolofuelRecentTransactionsQuery)
 
   useFetchCounterparties()
   const payTransaction = useOffer()
   const declineTransaction = useDecline()
-  const [modalTransaction, setModalTransaction] = useState()
-  const isTransactionsEmpty = isEmpty(transactions)
-
-  const pageTitle = `Inbox${isTransactionsEmpty ? '' : ` (${transactions.length})`}`
+  const [toggleModal, setToggleModal] = useState(null)
+  const [modalTransaction, setModalTransaction] = useState(null)
 
-  const showConfirmationModal = (transaction, action) => {
+  const showConfirmationModal = (transaction = {}, action = '') => {
     const modalTransaction = { ...transaction, action }
-    setModalTransaction(modalTransaction)
+    if (!isEmpty(transaction) && action !== '') return setModalTransaction(modalTransaction)
+    else return setToggleModal(true)
   }
 
-  return <PrimaryLayout headerProps={{ title: pageTitle }}>
+  const [actionsVisible, setActionsVisible] = useState(null)
+  const actionsClickWithTx = transaction => setActionsVisible(transaction)
```

Comment:
this is redundant
`actionsClickWithTx` is the same function as `setActionsVisible`